app/transport/rest/user.go

```diff
@@ -60,3 +70,41 @@ func (uh UserHandler) CreateUser() http.Handler {
 		uh.logger.Debugw("User successfully created", "ID", id)
 	})
 }
+
+// GetUserByID will handle user search.
+func (uh UserHandler) GetUserByID() http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		id := mux.Vars(req)["id"]
+
+		if _, err := uuid.Parse(id); err != nil {
+			uh.logger.Warnw("Invalid request", "ID", id)
+			WriteJSONResponse(w, http.StatusBadRequest, Response{Message: MsgBadRequest, Details: err.Error()}, uh.logger)
+
+			return
+		}
+
+		entUser, err := uh.usecase.GetUser(id)
+		if err != nil {
+			if errors.Is(err, globals.ErrNotFound) {
+				uh.logger.Debugw("No user found.", "ID", id)
+				WriteJSONResponse(w, http.StatusNotFound, Response{Message: MsgBadRequest,
+					Details: fmt.Sprintf("User with id: %s not found", id)}, uh.logger)
```

Comment:
It's a 404, not 400

Reply:
fixed .

Reply:
No, it should be 404, but the Message is `MsgBadRequest`, it should be a different message.

Reply:
But here already was the status "404" with the message `MsgNotFound` , Do you mean the message should be named as `MsgUserNotFound` whit value "User not found" ?

Reply:
Hm. I returned code to the state of the [commit](https://github.com/go-devs-ua/octagon/pull/29/commits/06ce460b2d6cf7982907460b7b14f4b7493fe80e)

Reply:
fix it

Reply:
fix it